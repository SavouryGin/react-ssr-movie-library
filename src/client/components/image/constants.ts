export const FALLBACK_IMAGE_PATH = 'client/assets/images/Fallback.jpg';
