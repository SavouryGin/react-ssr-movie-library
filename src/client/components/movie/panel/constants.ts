import { SelectEntity } from 'types/controls';
import { SortOrder, SortingValues } from 'enums/params';

export const sortOptions: SelectEntity[] = [
  {
    option: 'Most popular',
    value: SortingValues.MostPopular,
  },
  {
    option: 'Release (newest)',
    value: SortingValues.DateDesc,
    params: {
      sortBy: 'release_date',
      sortOrder: SortOrder.Desc,
    },
  },
  {
    option: 'Release (oldest)',
    value: SortingValues.DateAsc,
    params: {
      sortBy: 'release_date',
      sortOrder: SortOrder.Asc,
    },
  },
  {
    option: 'Title (A - Z)',
    value: SortingValues.TitleAsc,
    params: {
      sortBy: 'title',
      sortOrder: SortOrder.Asc,
    },
  },
  {
    option: 'Title (Z - A)',
    value: SortingValues.TitleDesc,
    params: {
      sortBy: 'title',
      sortOrder: SortOrder.Desc,
    },
  },
  {
    option: 'Rating (highest)',
    value: SortingValues.RatingDesc,
    params: {
      sortBy: 'vote_average',
      sortOrder: SortOrder.Desc,
    },
  },
  {
    option: 'Rating (lowest)',
    value: SortingValues.RatingAsc,
    params: {
      sortBy: 'vote_average',
      sortOrder: SortOrder.Asc,
    },
  },
  {
    option: 'Runtime (longest)',
    value: SortingValues.RuntimeDesc,
    params: {
      sortBy: 'runtime',
      sortOrder: SortOrder.Desc,
    },
  },
  {
    option: 'Runtime (shortest)',
    value: SortingValues.RuntimeAsc,
    params: {
      sortBy: 'runtime',
      sortOrder: SortOrder.Asc,
    },
  },
];
