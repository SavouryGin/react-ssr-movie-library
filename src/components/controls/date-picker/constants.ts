export const MAX_DATE = '2100-01-01';
export const MIN_DATE = '1901-01-01';
