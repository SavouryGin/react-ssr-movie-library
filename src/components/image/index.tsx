import React from 'react';
import classNames from 'classnames';
import { CommonProps } from 'types/basic';

type ImageProps = CommonProps & {
  path: string;
  altText: string;
};

const Image = ({ className, path, altText }: ImageProps) => {
  const imageClass = classNames('image', { [`${className}`]: !!className });

  return (
    <div className={imageClass}>
      <img src={path} className='mms_logo_img' alt={altText} role='img' />
    </div>
  );
};

export default Image;
