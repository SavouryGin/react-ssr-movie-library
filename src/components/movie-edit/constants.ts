import { Genre } from 'enums/genre';
import { MovieValues } from 'types/movies';

export const movieInitialValues: MovieValues = { title: '', url: '', genres: [], date: '', rating: 0, runtime: 0, overview: '' };

export const genreOptions = [
  {
    label: 'Action',
    value: Genre.Action,
  },
  {
    label: 'Adventure',
    value: Genre.Adventure,
  },
  {
    label: 'Drama',
    value: Genre.Drama,
  },
  {
    label: 'Documentary',
    value: Genre.Documentary,
  },
  {
    label: 'Comedy',
    value: Genre.Comedy,
  },
  {
    label: 'Music',
    value: Genre.Music,
  },
  {
    label: 'Crime',
    value: Genre.Crime,
  },
  {
    label: 'Horror',
    value: Genre.Horror,
  },
  {
    label: 'Biography',
    value: Genre.Biography,
  },
];
