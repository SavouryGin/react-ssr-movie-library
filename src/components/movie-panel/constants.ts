import { SelectEntity } from 'types/controls';
import { SortingValues } from 'enums/sorting-values';

export const sortOptions: SelectEntity[] = [
  {
    option: 'Release date',
    value: SortingValues.Date,
  },
  {
    option: 'Title',
    value: SortingValues.Title,
  },
];
