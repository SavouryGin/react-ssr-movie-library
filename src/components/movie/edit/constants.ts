import { Genre } from 'enums/genre';
import { MovieItem } from 'types/movies';

export const movieDefaultValues: MovieItem = { id: '', title: '', url: '', genres: [], date: '', rating: 0, runtime: 0, overview: '' };

export const genreOptions = [
  {
    label: 'Action',
    value: Genre.Action,
  },
  {
    label: 'Adventure',
    value: Genre.Adventure,
  },
  {
    label: 'Drama',
    value: Genre.Drama,
  },
  {
    label: 'Documentary',
    value: Genre.Documentary,
  },
  {
    label: 'Comedy',
    value: Genre.Comedy,
  },
  {
    label: 'Music',
    value: Genre.Music,
  },
  {
    label: 'Crime',
    value: Genre.Crime,
  },
  {
    label: 'Horror',
    value: Genre.Horror,
  },
  {
    label: 'Biography',
    value: Genre.Biography,
  },
];

export const inputLimits = {
  maxTextInputLength: 1000,
  minTextInputLength: 0,
  maxRating: 10,
  minRating: 0,
  maxRuntime: 1000,
  minRuntime: 0,
  ratingStep: 0.1,
  runtimeStep: 1,
};
