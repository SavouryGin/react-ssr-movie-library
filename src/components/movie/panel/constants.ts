import { SelectEntity } from 'types/controls';
import { SortingValues } from 'enums/sorting-values';

export const sortOptions: SelectEntity[] = [
  {
    option: 'Release (newest)',
    value: SortingValues.DateDesc,
    params: {
      sortBy: 'release_date',
      sortOrder: 'desc',
    },
  },
  {
    option: 'Release (oldest)',
    value: SortingValues.DateAsc,
    params: {
      sortBy: 'release_date',
      sortOrder: 'asc',
    },
  },
  {
    option: 'Title (A - Z)',
    value: SortingValues.TitleAsc,
    params: {
      sortBy: 'title',
      sortOrder: 'asc',
    },
  },
  {
    option: 'Title (Z - A)',
    value: SortingValues.TitleDesc,
    params: {
      sortBy: 'title',
      sortOrder: 'desc',
    },
  },
  {
    option: 'Rating (highest)',
    value: SortingValues.RatingDesc,
    params: {
      sortBy: 'vote_average',
      sortOrder: 'desc',
    },
  },
  {
    option: 'Rating (lowest)',
    value: SortingValues.RatingAsc,
    params: {
      sortBy: 'vote_average',
      sortOrder: 'asc',
    },
  },
  {
    option: 'Runtime (longest)',
    value: SortingValues.RuntimeDesc,
    params: {
      sortBy: 'runtime',
      sortOrder: 'desc',
    },
  },
  {
    option: 'Runtime (shortest)',
    value: SortingValues.RuntimeAsc,
    params: {
      sortBy: 'runtime',
      sortOrder: 'asc',
    },
  },
];
