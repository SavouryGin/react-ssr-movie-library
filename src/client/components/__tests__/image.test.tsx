import Image from 'components/image/index';
import React from 'react';
import userEvent from '@testing-library/user-event';
import { FALLBACK_IMAGE_PATH } from '../image/constants';
import { render, screen } from '@testing-library/react';

const testProps = {
  path: FALLBACK_IMAGE_PATH,
  altText: 'test alt text',
  onClickImage: jest.fn(),
};

describe('Image component:', () => {
  it('Should render the image', () => {
    render(<Image {...testProps} />);
    const image = screen.getByAltText(testProps.altText);
    expect(image).toBeInTheDocument();
  });

  it('Should match the snapshot', () => {
    const { asFragment } = render(<Image {...testProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Should call the passed onClickImage handler', () => {
    // arrange
    render(<Image {...testProps} />);
    const image = screen.getByRole('img');

    // act
    userEvent.click(image);

    // assert
    expect(testProps.onClickImage).toHaveBeenCalledTimes(1);
  });
});
