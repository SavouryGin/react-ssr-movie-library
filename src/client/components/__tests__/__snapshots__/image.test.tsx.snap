// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image component: Should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class=""
  >
    <img
      alt="test alt text"
      role="img"
      src="assets/images/Fallback.jpg"
    />
  </div>
</DocumentFragment>
`;
